File: python/Problem022.py
#871198282 in 0.012988805770874023
import string
import time

start = time.time()

f = open("../txt/p022_names.txt",'r')
names = f.read()
f.close()
names = names.replace('"',"")
names = names.split(",")
names.sort()
total = 0


for i in range(len(names)):
    temp = 0
    for j in names[i]:
        temp += ord(j.lower()) - 96
    total += temp * (i + 1)

print(total,"in",time.time()-start)

File: java/Problem022.java
package main;

import java.util.Arrays;

public class Problem022 {

	public Problem022() {
		String[] names = Utils.loadFile("p022_names").get(0).replace("\"", "").split(",");
		Arrays.sort(names);
		int sum = 0;
		for(int i = 0; i < names.length; i++) {
			sum += (alphabeticalValue(names[i]) * (i + 1));
		}
		System.out.println(sum);
	}
	
	public int alphabeticalValue(String s) {
		int sum = 0;
		char[] chars = s.toCharArray();
		for(char c : chars) {
			sum += ((int) c - 64);
		}
		return sum;
	}
	
	public static void main(String[] args) {
		new Problem022();
	}

}
